### PR TITLE
[#19908] locale in rest calls is always default locale

### PR DESCRIPTION
**Pull request by jwundrak**
### Description (*)

If there are multiple store views with different locales, always default store locale is present. For reproducing see #19908 . 

### Fixed Issues (if relevant)

#19908 

### Manual testing scenarios (*)
1. Configure stores. I use Store-View "default" with locale en_US and StoreView "second" with locale de_DE
2. Use something, that returns magento phrase (I wrote a test module here: https://github.com/jwundrak/magento2-api-locale )
3. Call rest API (with the module /rest/second/V1/localeTest)

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=19913) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @jwundrak. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Webapi/Controller/PathProcessor.php [hunk 1]
```diff
@@ -21,12 +21,21 @@ class PathProcessor
...
+
     /**
      * @param \Magento\Store\Model\StoreManagerInterface $storeManager
+     * @param \Magento\Framework\Locale\ResolverInterface $localeResolver
      */
-    public function __construct(\Magento\Store\Model\StoreManagerInterface $storeManager)
-    {
+    public function __construct(
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        \Magento\Framework\Locale\ResolverInterface $localeResolver
```
Comment (sidolov) [inline app/code/Magento/Webapi/Controller/PathProcessor.php]: Please, inject new dependencies according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

File: app/code/Magento/Webapi/Test/Unit/Controller/PathProcessorTest.php [hunk 2]
```diff
@@ -57,10 +73,10 @@ public function testAllStoreCode($storeCodeInPath, $storeCodeSet, $setCurrentSto
     public function processPathDataProvider()
     {
         return [
-            'All store code' => ['all', Store::ADMIN_CODE],
-            'Default store code' => ['', 'default', 0],
-            'Arbitrary store code' => [$this->arbitraryStoreCode, $this->arbitraryStoreCode],
-            'Explicit default store code' => ['default', 'default']
+            'All store code'              => ['all', Store::ADMIN_CODE],
```
Comment (sidolov) [inline app/code/Magento/Webapi/Test/Unit/Controller/PathProcessorTest.php]: Please, revert indentation

File: app/code/Magento/Webapi/Test/Unit/Controller/PathProcessorTest.php [hunk 3]
```diff
@@ -24,29 +27,42 @@ class PathProcessorTest extends \PHPUnit\Framework\TestCase
...
      * @param string $storeCodeSet
-     * @param int $setCurrentStoreCallCtr
+     * @param int    $setCurrentStoreCallCtr
      */
     public function testAllStoreCode($storeCodeInPath, $storeCodeSet, $setCurrentStoreCallCtr = 1)
     {
-        $storeCodeInPath = !$storeCodeInPath ?: '/' . $storeCodeInPath; // add leading slash if store code not empty
-        $inPath = 'rest' . $storeCodeInPath . $this->endpointPath;
+        $storeCodeInPath = !$storeCodeInPath ? : '/' . $storeCodeInPath; // add leading slash if store code not empty
+        $inPath          = 'rest' . $storeCodeInPath . $this->endpointPath;
```
Comment (sidolov) [inline app/code/Magento/Webapi/Test/Unit/Controller/PathProcessorTest.php]: Please, revert indentation

File: dev/tests/integration/testsuite/Magento/Webapi/Controller/PathProcessorTest.php [hunk 4]
```diff
@@ -15,6 +15,11 @@ class PathProcessorTest extends \PHPUnit\Framework\TestCase
      */
     protected $storeManager;
 
+    /**
+     * @var \Magento\Framework\Locale\ResolverInterface::class
+     */
+    protected $localeResolver;
```
Comment (sidolov) [inline dev/tests/integration/testsuite/Magento/Webapi/Controller/PathProcessorTest.php]: Please, make this property private

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Webapi/Controller/PathProcessor.php (hunk 1 above)
Comment (jwundrak) [reply app/code/Magento/Webapi/Controller/PathProcessor.php]: @sidolov Sorry, done

File: app/code/Magento/Webapi/Test/Unit/Controller/PathProcessorTest.php (hunk 2 above)
Comment (jwundrak) [reply app/code/Magento/Webapi/Test/Unit/Controller/PathProcessorTest.php]: @sidolov Sorry, done

File: app/code/Magento/Webapi/Test/Unit/Controller/PathProcessorTest.php (hunk 3 above)
Comment (jwundrak) [reply app/code/Magento/Webapi/Test/Unit/Controller/PathProcessorTest.php]: @sidolov Sorry, done

File: dev/tests/integration/testsuite/Magento/Webapi/Controller/PathProcessorTest.php (hunk 4 above)
Comment (jwundrak) [reply dev/tests/integration/testsuite/Magento/Webapi/Controller/PathProcessorTest.php]: @sidolov Sorry, done

**Review by sidolov — APPROVED** [review]

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-4674](https://jira.corp.magento.com/browse/ENGCOM-4674) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@jwundrak thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Pull request merged**